File: FurnitureStore/FurnitureStore.Models/Report/Report.cs

using FurnitureStore.Models.Account;
using FurnitureStore.Models.Enums;

namespace FurnitureStore.Models.Report
{
    public class Report
    {
        public long Id { get; set; }
        public DateTime GenerationDate { get; set; }
        public Month Month { get; set; }
        public ReportType ReportType { get; set; }
        public int Year { get; set; }
        public string Content { get; set; }
        public string AdminId { get; set; }
        public UserResponse Admin { get; set; }
        public string CustomerId { get; set; }
        public UserResponse Customer { get; set; }
    }
}

File: FurnitureStore/UI/furniturestore_admin/lib/models/report/report.dart
import 'package:furniturestore_admin/models/account/account.dart';
import 'package:json_annotation/json_annotation.dart';

part 'report.g.dart';

@JsonSerializable()
class ReportModel {
  int? id;
  DateTime? generationDate;
  int? month;
  int? year;
  String? content;
  int? admin;
  String? customerId;
  List<AccountModel>? customer;
  int? reportType;

  ReportModel({
    this.id,
    this.generationDate,
    this.month,
    this.year,
    this.content,
    this.admin,
    this.customerId,
    this.customer,
    this.reportType,
  });

  factory ReportModel.fromJson(Map<String, dynamic> json) =>
      _$ReportModelFromJson(json);
  Map<String, dynamic> toJson() => _$ReportModelToJson(this);
}
